packages/rest/src/router/metadata.ts

```diff
@@ -72,10 +72,65 @@ export function api(spec: ControllerSpec) {
       'The @api decorator can be applied to constructors only.',
     );
     const apiSpec = resolveControllerSpec(constructor, spec);
-    Reflector.defineMetadata(API_SPEC_KEY, apiSpec, constructor);
+    const key = constructor.prototype._server
+      ? `${API_SPEC_KEY}:${constructor.prototype._server}`
+      : API_SPEC_KEY;
+    Reflector.defineMetadata(key, apiSpec, constructor);
   };
 }
 
+/**
+ * A decorator to mark a controller as belonging to a specific server.
+ * ```ts
+ * // In your application...
+ * class MyApplication extends Application {
+ *   constructor() {
+ *     super();
+ *     this.server(RestServer, 'foo');
+ *   }
+ * }
+ *
+ * // In one of your controllers
+ * @server('foo')
+ * class MyController {
+ *  // ...
+ * }
+ * ```
+ * @param name The name of the server bound to your application context.
+ */
+export function server(name: string) {
+  return function(constructor: Constructor<any>) {
+    const endpoints = Reflector.getOwnMetadata(ENDPOINTS_KEY, constructor);
+    if (endpoints) {
+      rescopeMetadata(ENDPOINTS_KEY, name, constructor, endpoints);
+    }
+    const spec = getControllerSpec(constructor);
```

Comment:
It might be too early to call `getControllerSpec` as there are cases that not all decorators have been processed.

Reply:
All of the inner operation decorators on the class are processed before the `@server` decorator runs, and this only changes the target for the metadata. Any subsequent changes to that metadata reference would be made using the new keys.

If that's not the case, I'm not sure _where_ those potential changes are being made.